common/maven-url-handler/src/main/java/io/helidon/build/maven/url/MavenURLHandler.java

```diff
@@ -0,0 +1,464 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.build.maven.url;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLStreamHandler;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+/**
+ * Maven Url Stream Handler.
+ */
+public class MavenURLHandler extends URLStreamHandler {
+
+    @Override
+    protected URLConnection openConnection(URL url) throws IOException {
+        MavenResolver resolver = new MavenResolver();
+        return new Connection(url, resolver);
+    }
+
+    @Override
+    protected void parseURL(URL u, String spec, int start, int limit) {
+        super.setURL(u, null, null, 0,  null, null, spec.substring(4), null, null);
+    }
+
+    static class MavenResolver {
```

Comment:
I did I tell you to inline it, but if you want you can move it out to a separate class (since the connection class is not inline, it may be more consistent to have them both at the same level). If you do please use a name that's rather specific (e.g. MavenFileResolver) and make it package private.